geopandas/sindex.py

```diff
@@ -168,6 +168,25 @@ def query_bulk(self, geometry, predicate=None, sort=False):
         """
         raise NotImplementedError
 
+    def nearest(self, geometry):
+        """
+        Returns the nearest geometry to the input geometries.
+
+        Parameters
+        ----------
+        geometry : {shapely geometry, GeoSeries, GeometryArray, numpy.array of PyGEOS
+            geometries}
+            A single shapely geometry, one of the GeoPandas geometry iterables
+            (GeoSeries, GeometryArray), or a numpy array of PyGEOS geometries to query
+            against the spatial index.
+
+        Returns
+        -------
+        matches : ndarray of shape (n_input_geometries, )
```

Comment:
Same output structure as `query_bulk`:
```
        ndarray with shape (2, n)
            The first subarray contains input geometry integer indexes.
            The second subarray contains spatial index geometry integer indexes.
```

Reply:
Can you also please add a brief example similar to `query_bulk`?

I'm not sure if the doctesting will break due to pygeos version though.

Reply:
Oops, I've fixed the comment. Yes, I have added an example, hope it gets through doctesting.